### PR TITLE
Port tsdb to use pkg/labels.

### PR DESCRIPTION
**Pull request by tomwilkie**
This removes a bunch of conversions and duplicate code.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: pkg/labels/labels.go [hunk 1]
```diff
@@ -201,6 +201,23 @@ func (ls Labels) Has(name string) bool {
 	return false
 }
 
+// WithoutEmpty returns the labelset without empty labels.
+// May return the same labelset.
+func (ls Labels) WithoutEmpty() Labels {
+	for _, v := range ls {
+		if v.Value == "" {
```
Comment (bwplotka) [inline pkg/labels/labels.go]: minor: 
```
if v.Value != "" {
   continue
}
```
.. would make a complex path the "main one" in this function (aka filter boring cases first) (:

File: pkg/labels/matcher.go [hunk 2]
```diff
@@ -86,3 +95,18 @@ func (m *Matcher) Matches(s string) bool {
...
+	case MatchEqual:
+		return NewMatcher(MatchNotEqual, m.Name, m.Value)
+	case MatchNotEqual:
+		return NewMatcher(MatchEqual, m.Name, m.Value)
+	case MatchRegexp:
+		return NewMatcher(MatchNotRegexp, m.Name, m.Value)
+	case MatchNotRegexp:
+		return NewMatcher(MatchRegexp, m.Name, m.Value)
+	}
+	panic("labels.Matcher.Matches: invalid match type")
```
Comment (bwplotka) [inline pkg/labels/matcher.go]:
```suggestion
	panic("labels.Matcher.Inverse: invalid match type")
```

Comment (bwplotka) [reply pkg/labels/matcher.go]: Also not sure why panic if we have error handling piped already (:

Comment (brian-brazil) [reply pkg/labels/matcher.go]: It's a coding bug, so a panic isn't inappropriate.

**Pull request merged**